### ISSUE TITLE
codingStandardsChangeSetting settings not working with namespaces

### ISSUE THREAD

**Issue by evilebottnawi** [squizlabs/PHP_CodeSniffer#791]
if use namespaces in sniffs and write test with directive `codingStandardsChangeSetting`, this directive not working, not change value of sniff

**Comment by gsherwood** [comment]
Can you please give me an example sniff and file being checked that isn't working?

Namespaces shouldn't have any effect on sniff settings because settings use the sniff code (standard.category.sniffname) and not a fully qualified class name. So also make sure you have the class name correct I guess.

**Comment by evilebottnawi** [comment]
@gsherwood 
Sorry for long awaiting.

Sniff:
```php
namespace MySuperStandard\Sniffs\Strings;

class DoubleQuoteUsageSniff implements \PHP_CodeSniffer_Sniff
{
    public $someVariable = 0;
}
```
Test .inc:
```php
// @codingStandardsChangeSetting MySuperStandard.Strings.DoubleQuoteUsage someVariable 1
```
And it is not because in `File.php` on line 505 (https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/File.php#L505) variable `$listenerClass` equal `MySuperStandard_Sniffs_Strings_DoubleQuoteUsageSniff`, but `$listeners  = $this->phpcs->getSniffs()` return array
```php
Array
(
    [MySuperStandard\Sniffs\Strings\DoubleQuoteUsageSniff] => MySuperStandard\Sniffs\Strings\DoubleQuoteUsageSniff Object
        (
            [someVariable] => 0
        )
)
``` and in `PHP_CodeSniffer.php` on line 1475 (https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer.php#L1475) check name sniff in array, without namespaces. I tried to explain as much as possible.

**Comment by evilebottnawi** [comment]
@gsherwood bump

**Comment by gsherwood** [comment]
Yeah sorry. I've got this on my todo list but haven't had time to look at it.

**Comment by evilebottnawi** [comment]
@gsherwood well I'll see what I can do for this. Maybe change labels on issue?

**Comment by evilebottnawi** [comment]
@gsherwood expected implementation or better PR?

**Comment by gsherwood** [comment]
I'm really sorry. I haven't had a spare moment to even look at this yet. The notification email is still sitting in my stared list in gmail though - I promise I'll get to it when work slows down.

**Comment by gsherwood** [comment]
I finally got time to look at this :)

The 2.x version does not fully support using namespaces for sniffs, while the 3.x version *only* supports using namespaces (they will be a new requirement when 3.0 is released). So I back-ported a new array from 3.x and got this working. It's a bit of a hack, but it's done in a nicer way in the 3.0 branch and that will eventually become the code that properly supports namespaces, so I'm ok withe the temporary hack.

Thanks for being so patient.
